api/sign.py

```
@@ -0,0 +1,164 @@
+
+    def _do_code(self, active_id, name, info):
+        code = input("请输入签到码: ").strip()
+        if not code:
+            return None
+        result = self.general_sign(active_id, name + f"&code={code}")
+        msg = "签到成功" if result == "success" else f"签到失败: {result}"
+        logger.info(f"[签到码] {msg}")
+        return msg
+
+    def monitor(self, courses, sign_config, interval=30):
+        logger.info(f"签到监控已启动，每 {interval} 秒检查一次...")
```

Ruff (0.16.2)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)